src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/UpdateInductionTest.kt

```diff
@@ -0,0 +1,307 @@
+        id = persistedInduction.skillsAndInterests!!.id!!,
+        skills = setOf(PersonalSkill.COMMUNICATION),
+        skillsOther = null,
+        personalInterests = setOf(PersonalInterest.CRAFTS),
+        personalInterestsOther = null,
+      ),
+      qualificationsAndTraining = aValidUpdateEducationAndQualificationsRequest(
+        id = persistedInduction.qualificationsAndTraining!!.id!!,
+        educationLevel = HighestEducationLevel.PRIMARY_SCHOOL,
+        qualifications = emptySet(),
+        additionalTraining = setOf(TrainingType.CSCS_CARD),
+        additionalTrainingOther = null,
+      ),
+      inPrisonInterests = aValidUpdatePrisonWorkAndEducationRequest(
+        id = persistedInduction.inPrisonInterests!!.id!!,
+        inPrisonWork = setOf(InPrisonWorkType.PRISON_LIBRARY),
+        inPrisonWorkOther = null,
+        inPrisonEducation = setOf(InPrisonTrainingType.WELDING_AND_METALWORK),
+        inPrisonEducationOther = null,
+      ),
+    )
+
+    val expectedWorkExperience = aValidPreviousWorkResponse(
+      id = persistedInduction.workExperience!!.id!!,
+      typeOfWorkExperience = setOf(WorkType.CONSTRUCTION),
+      typeOfWorkExperienceOther = null,
+      workExperience = setOf(
+        aValidWorkExperienceResource(
+          typeOfWorkExperience = WorkType.CONSTRUCTION,
+          otherWork = null,
+          role = "Bricklayer",
+          details = "General brick work",
+        ),
+      ),
+      workInterests = aValidWorkInterests(
+        id = persistedInduction.workExperience!!.workInterests!!.id,
+        workInterests = setOf(WorkType.SPORTS),
+        workInterestsOther = null,
+        particularJobInterests = setOf(
+          WorkInterestDetail(
+            workInterest = WorkType.SPORTS,
+            role = "Football coach",
+          ),
+        ),
+      ),
+      modifiedBy = "auser_gen",
+    )
+    val expectedSkillsAndInterests = aValidSkillsAndInterestsResponse(
+      id = persistedInduction.skillsAndInterests!!.id!!,
+      skills = setOf(PersonalSkill.COMMUNICATION),
+      skillsOther = null,
+      personalInterests = setOf(PersonalInterest.CRAFTS),
+      personalInterestsOther = null,
+      modifiedBy = "auser_gen",
+    )
+    val expectedQualificationsAndTraining = aValidEducationAndQualificationsResponse(
+      id = persistedInduction.qualificationsAndTraining!!.id!!,
+      educationLevel = HighestEducationLevel.PRIMARY_SCHOOL,
+      qualifications = emptySet(),
+      additionalTraining = setOf(TrainingType.CSCS_CARD),
+      additionalTrainingOther = null,
+      modifiedBy = "auser_gen",
+    )
+    val expectedInPrisonInterests = aValidPrisonWorkAndEducationResponse(
+      id = persistedInduction.inPrisonInterests!!.id!!,
+      inPrisonWork = setOf(InPrisonWorkType.PRISON_LIBRARY),
+      inPrisonWorkOther = null,
+      inPrisonEducation = setOf(InPrisonTrainingType.WELDING_AND_METALWORK),
+      inPrisonEducationOther = null,
+      modifiedBy = "auser_gen",
+    )
```

Comment:
I appreciate this is a large given atm. I'm planning on splitting this up into different tests - e.g. separates ones for updating skillsAndInterests (with and without mandatory fields), but as I said before, I want to address the cardinality separately. For now, I just want to make sure that everything gets updated as expected (i.e. when all child objects are provided in the update request).

Reply:
Sounds like a sensible plan 👍